// angular-news-api/bundles/angular-news-api.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('newsapi'), require('rxjs'), require('@angular/core')) :
    typeof define === 'function' && define.amd ? define('angular-news-api', ['exports', 'newsapi', 'rxjs', '@angular/core'], factory) :
    (factory((global['angular-news-api'] = {}),global.NewsAPI,global.rxjs,global.ng.core));
}(this, (function (exports,NewsAPI,rxjs,i0) { 'use strict';

    NewsAPI = NewsAPI && NewsAPI.hasOwnProperty('default') ? NewsAPI['default'] : NewsAPI;

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    /** @type {?} */
    var NewsApiConfigService = new i0.InjectionToken('NewsApiConfig');

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var NewsApiService = /** @class */ (function () {
        function NewsApiService(newsApiConfigService) {
            this.newsApiConfigService = newsApiConfigService;
            this.newsApi = new NewsAPI(this.newsApiConfigService.key);
        }
        /**
         * @param {?} config
         * @return {?}
         */
        NewsApiService.prototype.topHeadlines = /**
         * @param {?} config
         * @return {?}
         */
            function (config) {
                return rxjs.from(this.newsApi.v2.topHeadlines(config));
            };
        /**
         * @param {?} config
         * @return {?}
         */
        NewsApiService.prototype.everything = /**
         * @param {?} config
         * @return {?}
         */
            function (config) {
                return rxjs.from(this.newsApi.v2.everything(config));
            };
        /**
         * @param {?} config
         * @return {?}
         */
        NewsApiService.prototype.sources = /**
         * @param {?} config
         * @return {?}
         */
            function (config) {
                return rxjs.from(this.newsApi.v2.sources(config));
            };
        NewsApiService.decorators = [
            { type: i0.Injectable, args: [{
                        providedIn: 'root'
                    },] }
        ];
        /** @nocollapse */
        NewsApiService.ctorParameters = function () {
            return [
                { type: undefined, decorators: [{ type: i0.Inject, args: [NewsApiConfigService,] }] }
            ];
        };
        /** @nocollapse */ NewsApiService.ngInjectableDef = i0.defineInjectable({ factory: function NewsApiService_Factory() { return new NewsApiService(i0.inject(NewsApiConfigService)); }, token: NewsApiService, providedIn: "root" });
        return NewsApiService;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var NgnewsModule = /** @class */ (function () {
        function NgnewsModule() {
        }
        /**
         * @param {?} newsApiConfig
         * @return {?}
         */
        NgnewsModule.forRoot = /**
         * @param {?} newsApiConfig
         * @return {?}
         */
            function (newsApiConfig) {
                return {
                    ngModule: NgnewsModule,
                    providers: [
                        NewsApiService,
                        {
                            provide: NewsApiConfigService, useValue: newsApiConfig
                        }
                    ]
                };
            };
        NgnewsModule.decorators = [
            { type: i0.NgModule, args: [{},] }
        ];
        return NgnewsModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */

    exports.NewsApiService = NewsApiService;
    exports.NgnewsModule = NgnewsModule;
    exports.ɵa = NewsApiConfigService;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
