// angular-news-api/esm2015/lib/interfaces/newsapi.models.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * @record
 */
export function NewsApiKeyConfig() { }
if (false) {
    /** @type {?} */
    NewsApiKeyConfig.prototype.key;
}
/**
 * @record
 */
export function TopHeadlinesResponse() { }
if (false) {
    /** @type {?} */
    TopHeadlinesResponse.prototype.status;
    /** @type {?} */
    TopHeadlinesResponse.prototype.totalResults;
    /** @type {?} */
    TopHeadlinesResponse.prototype.articles;
}
/**
 * @record
 */
export function EverythingResponse() { }
if (false) {
    /** @type {?} */
    EverythingResponse.prototype.status;
    /** @type {?} */
    EverythingResponse.prototype.totalResults;
    /** @type {?} */
    EverythingResponse.prototype.articles;
}
/**
 * @record
 */
export function SourcesResponse() { }
if (false) {
    /** @type {?} */
    SourcesResponse.prototype.status;
    /** @type {?} */
    SourcesResponse.prototype.sources;
}
/**
 * @record
 */
export function Article() { }
if (false) {
    /** @type {?} */
    Article.prototype.source;
    /** @type {?|undefined} */
    Article.prototype.author;
    /** @type {?} */
    Article.prototype.title;
    /** @type {?} */
    Article.prototype.description;
    /** @type {?} */
    Article.prototype.url;
    /** @type {?} */
    Article.prototype.urlToImage;
    /** @type {?} */
    Article.prototype.publishedAt;
    /** @type {?|undefined} */
    Article.prototype.content;
}
/**
 * @record
 */
export function Source() { }
if (false) {
    /** @type {?} */
    Source.prototype.id;
    /** @type {?} */
    Source.prototype.name;
    /** @type {?} */
    Source.prototype.description;
    /** @type {?} */
    Source.prototype.url;
    /** @type {?} */
    Source.prototype.category;
    /** @type {?} */
    Source.prototype.language;
    /** @type {?} */
    Source.prototype.country;
}
/**
 * @record
 */
export function TopHeadlinesConfig() { }
if (false) {
    /** @type {?|undefined} */
    TopHeadlinesConfig.prototype.sources;
    /** @type {?|undefined} */
    TopHeadlinesConfig.prototype.q;
    /** @type {?|undefined} */
    TopHeadlinesConfig.prototype.category;
    /** @type {?|undefined} */
    TopHeadlinesConfig.prototype.language;
    /** @type {?|undefined} */
    TopHeadlinesConfig.prototype.country;
}
/**
 * @record
 */
export function EverythingConfig() { }
if (false) {
    /** @type {?|undefined} */
    EverythingConfig.prototype.q;
    /** @type {?|undefined} */
    EverythingConfig.prototype.sources;
    /** @type {?|undefined} */
    EverythingConfig.prototype.domains;
    /** @type {?|undefined} */
    EverythingConfig.prototype.from;
    /** @type {?|undefined} */
    EverythingConfig.prototype.to;
    /** @type {?|undefined} */
    EverythingConfig.prototype.language;
    /** @type {?|undefined} */
    EverythingConfig.prototype.sortBy;
    /** @type {?|undefined} */
    EverythingConfig.prototype.page;
}
/**
 * @record
 */
export function SourcesConfig() { }
if (false) {
    /** @type {?|undefined} */
    SourcesConfig.prototype.category;
    /** @type {?|undefined} */
    SourcesConfig.prototype.language;
    /** @type {?|undefined} */
    SourcesConfig.prototype.country;
}
