// angular-news-api/fesm5/angular-news-api.js
import NewsAPI from 'newsapi';
import { from } from 'rxjs';
import { InjectionToken, Injectable, Inject, NgModule, defineInjectable, inject } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/** @type {?} */
var NewsApiConfigService = new InjectionToken('NewsApiConfig');

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var NewsApiService = /** @class */ (function () {
    function NewsApiService(newsApiConfigService) {
        this.newsApiConfigService = newsApiConfigService;
        this.newsApi = new NewsAPI(this.newsApiConfigService.key);
    }
    /**
     * @param {?} config
     * @return {?}
     */
    NewsApiService.prototype.topHeadlines = /**
     * @param {?} config
     * @return {?}
     */
    function (config) {
        return from(this.newsApi.v2.topHeadlines(config));
    };
    /**
     * @param {?} config
     * @return {?}
     */
    NewsApiService.prototype.everything = /**
     * @param {?} config
     * @return {?}
     */
    function (config) {
        return from(this.newsApi.v2.everything(config));
    };
    /**
     * @param {?} config
     * @return {?}
     */
    NewsApiService.prototype.sources = /**
     * @param {?} config
     * @return {?}
     */
    function (config) {
        return from(this.newsApi.v2.sources(config));
    };
    NewsApiService.decorators = [
        { type: Injectable, args: [{
                    providedIn: 'root'
                },] }
    ];
    /** @nocollapse */
    NewsApiService.ctorParameters = function () { return [
        { type: undefined, decorators: [{ type: Inject, args: [NewsApiConfigService,] }] }
    ]; };
    /** @nocollapse */ NewsApiService.ngInjectableDef = defineInjectable({ factory: function NewsApiService_Factory() { return new NewsApiService(inject(NewsApiConfigService)); }, token: NewsApiService, providedIn: "root" });
    return NewsApiService;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var NgnewsModule = /** @class */ (function () {
    function NgnewsModule() {
    }
    /**
     * @param {?} newsApiConfig
     * @return {?}
     */
    NgnewsModule.forRoot = /**
     * @param {?} newsApiConfig
     * @return {?}
     */
    function (newsApiConfig) {
        return {
            ngModule: NgnewsModule,
            providers: [
                NewsApiService,
                {
                    provide: NewsApiConfigService, useValue: newsApiConfig
                }
            ]
        };
    };
    NgnewsModule.decorators = [
        { type: NgModule, args: [{},] }
    ];
    return NgnewsModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { NewsApiService, NgnewsModule, NewsApiConfigService as ɵa };
