// angular-news-api/esm2015/lib/services/news-api.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Injectable, Inject } from '@angular/core';
import NewsAPI from 'newsapi';
import { NewsApiConfigService } from './news-api-config.service';
import { from } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "./news-api-config.service";
export class NewsApiService {
    /**
     * @param {?} newsApiConfigService
     */
    constructor(newsApiConfigService) {
        this.newsApiConfigService = newsApiConfigService;
        this.newsApi = new NewsAPI(this.newsApiConfigService.key);
    }
    /**
     * @param {?} config
     * @return {?}
     */
    topHeadlines(config) {
        return from(this.newsApi.v2.topHeadlines(config));
    }
    /**
     * @param {?} config
     * @return {?}
     */
    everything(config) {
        return from(this.newsApi.v2.everything(config));
    }
    /**
     * @param {?} config
     * @return {?}
     */
    sources(config) {
        return from(this.newsApi.v2.sources(config));
    }
}
NewsApiService.decorators = [
    { type: Injectable, args: [{
                providedIn: 'root'
            },] }
];
/** @nocollapse */
NewsApiService.ctorParameters = () => [
    { type: undefined, decorators: [{ type: Inject, args: [NewsApiConfigService,] }] }
];
/** @nocollapse */ NewsApiService.ngInjectableDef = i0.defineInjectable({ factory: function NewsApiService_Factory() { return new NewsApiService(i0.inject(i1.NewsApiConfigService)); }, token: NewsApiService, providedIn: "root" });
if (false) {
    /**
     * @type {?}
     * @private
     */
    NewsApiService.prototype.newsApi;
    /**
     * @type {?}
     * @private
     */
    NewsApiService.prototype.newsApiConfigService;
}
